### DISCUSSION TITLE
Question(s) regarding Quadlet and Podman 4.4.1

### DISCUSSION THREAD

**Discussion by mcdent** [containers/podman#18897] [Q&A]
Hi, I'm moving my Podman containers on RHEL 9.2 to the newer Quartet way of doing things. 
One reason was to take advantage of being able to update them more easily.
I've managed to get one container working but there are a couple of issues. I'd like to iron these out before I continue with the rest 🤞

1) if you add the `Label="io.containers.autoupdate=registry"` to the unit file,  you need to specify the fully qualified path to the image. It seems no matter what I try, e.g. `hub.docker.com/r/louislam/uptime-kuma:latest` it does not work.

2) If I forget number 1 for now, remove the `Label="io.containers.autoupdate=registry"` and set `Image=louislam/uptime-kuma:latest` I am able to manually start the container from this Quartet file:-

```
[Unit]
Description=Uptime Kuma
After=local-fs.target

[Container]
Image=louislam/uptime-kuma:latest
#Image=hub.docker.com/r/louislam/uptime-kuma:latest
Volume=/home/mike/docker/uptime-kuma/run/data:/app/data
PublishPort=3001:3001
ContainerName=uptime-kuma
#Label="io.containers.autoupdate=registry"
Exec=

[Install]
# Start by default on boot
WantedBy=multi-user.target default.target
```

The process I used was:-
`systemctl --user daemon-reload`
`systemctl --user status mysleep.service`
And the service & container start fine. However if I reboot, the service/container does not start.
I've been picking at the following couple of websites for help:-
https://www.redhat.com/sysadmin/quadlet-podman
https://docs.podman.io/en/latest/markdown/podman-systemd.unit.5.html

Thanks for any pointers/help!
Mike
🙏

**Comment by eriksjolund** [comment]
I think you mean _Quadlet_  instead of _Quartet_ 

 Fully qualified paths for images at Dockerhub start with _docker.io/_

Try replacing
_hub.docker.com/r/louislam/uptime-kuma:latest_
with
_docker.io/louislam/uptime-kuma:latest_

**Reply by mcdent** [reply]
You are indeed correct! Quadlet :) My bad.
Thanks that has fixed the issue with the `Image` name.

Any ideas what I have missed which prevents the container starting at boot? Oddly the example I initially tested with from 
https://www.redhat.com/sysadmin/quadlet-podman , whicih just runs a sleep 1000 seems to start fine.
```
[Unit]
Description=The sleep container
After=local-fs.target

[Container]
Image=registry.access.redhat.com/ubi9-minimal:latest
Exec=sleep 1000

[Install]
# Start by default on boot
WantedBy=multi-user.target default.target
```

**Reply by eriksjolund** [reply]
Maybe you need to run

`loginctl enable-linger yourusername`

or 

`sudo loginctl enable-linger yourusername`

**Reply by mcdent** [reply]
Thanks, I already had that set. I have other Podman containers starting at boot and in fact the test one I posted above starts at boot just fine. I'll check the logs after another boot.

**Reply by eriksjolund** [reply]
I don't know how Quadlet treats empty values.

I see you use 
```
Exec=
```

Probably it's better to remove that line or to set the value to the command you would like to be run.

**Reply by mcdent** [reply]
Ok, I actually sorted this and it was an issue on my side.

It appears as I had an 'old' container and service file in `~/.config/systemd/user/` even though it was `disabled` it appeared to confilct with the new unit file of the same name `uptime-kuma`.
I tested this first by changing the `ContainerName=uptime-kuma` to `ContainerName=uptimekuma` and it started as expected at boot. 
I then removed the older service file to stop the errors I was seeing in the journal at boot:-

- `systemctl --user disable container-uptime-kuma.service`
- `systemctl --user stop container-uptime-kuma.service`
- `rm ~/.config/systemd/user/container-uptime-kuma.service`
- `systemctl --user daemon-reload`

So the moral of this story is to fully remove your old stype container service files before setting up the new Quadlet unit file.
